core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -598,12 +598,18 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::Cuda> {
         Kokkos::Impl::cuda_get_opt_block_size<FunctorType, LaunchBounds>(
             m_policy.space().impl_internal_space_instance(), attr, m_functor, 1,
             0, 0);
-    const dim3 block(1, block_size, 1);
-    const dim3 grid(
+    dim3 block(1, block_size, 1);
+    dim3 grid(
         std::min(
             typename Policy::index_type((nwork + block.y - 1) / block.y),
             typename Policy::index_type(cuda_internal_maximum_grid_count())),
         1, 1);
+#ifdef KOKKOS_IMPL_DEBUG_CUDA_SERIAL_EXECUTION
+    if (Kokkos::Impl::CudaInternal::cuda_use_serial_execution()) {
+      block = dim3(1, 1, 1);
+      grid  = dim3(1, 1, 1);
+    }
+#endif
```

Comment:
I think it is bad form to drop the const for convenience but I won't block if you don't want to fix.

Reply:
I thought about this, dropping the const results in less code duplication in a piece of code which passed experience shows is fragile. I felt that is the safer path to take.